inc/Engine/Media/AboveTheFold/Frontend/Controller.php

```diff
@@ -146,9 +146,17 @@ private function set_fetchpriority( $lcp, string $html ): string {
 			return $html;
 		}
 
-		$url  = preg_quote( $lcp->src, '/' );
+		$url     = preg_quote( $lcp->src, '/' );
+		$pattern = '#<img(?:[^>]*?\s+)?src=["\']' . $url . '["\'](?:\s+[^>]*?)?>#';
+		if ( wp_http_validate_url( $lcp->src ) && ! $this->is_external_file( $lcp->src ) ) {
+			$url = get_rocket_parse_url( $lcp->src );
```

Comment:
So here we only need the path? 

```suggestion
			$path = wp_parse_url( $lcp->src, PHP_URL_PATH );
```